### DISCUSSION TITLE
"static" hostname

### DISCUSSION THREAD

**Discussion by homonto** [HomeSpan/HomeSpan#444] [Ideas]
Hi, is there any possibility to set the static hostname, string based or even MAC based - anything that can be known in advance.
Today the hostname can only be modified by prefix but then there are digits from the sky ;-)
Reason being: I have added OTA web page where one can directly upload the bin file (without Arduino) but finding the hostname is kind of difficult without the serial port.
And also IP is dynamically assigned.

I would like to set it manually i.e. instead of:
`void setHostNameSuffix(const char *suffix)`

to be (or also to be):
`void setHostName(const char *hostname)`

is it possible?

thx

**Comment by HomeSpan** [comment]
You need to set the third argument of homeSpan.begin() to whatever static name you want, and then use homeSpan.setHostNameSuffix() to either set a new suffix, or just set to an empty string "".  See the API for an example of how these interact.

**Reply by homonto** [reply]
I missed the "suffix" part - now it is perfecto - thank you ;-)
` HostName:      hub1.local:80`
